libraries/FS/src/vfs_api.cpp

```diff
@@ -476,6 +476,36 @@ FileImplPtr VFSFileImpl::openNextFile(const char* mode)
     return std::make_shared<VFSFileImpl>(_fs, name.c_str(), mode);
 }
 
+boolean VFSFileImpl::seekDir(long position){
+    if(!_d){
+        return false;
+    }
+    seekdir(_d, position);
+    return true;
+}
+
+
+String VFSFileImpl::getNextFileName()
+{
+    if (!_isDirectory || !_d) {
+        return "";
+    }
+    struct dirent *file = readdir(_d);
+    if (file == NULL) {
+        return "";
+    }
+    if (file->d_type != DT_REG && file->d_type != DT_DIR) {
+        return "";
+    }
+    String fname = String(file->d_name);
+    String name = String(_path);
+    if (!fname.startsWith("/") && !name.endsWith("/")) {
+        name += "/";
+    }
+    name += fname;
+    return name;
```

Comment:
just a small hint, this could be reduced to avoid a couple of extra copy's

```c++
    String fname(file->d_name);
    if (!fname.startsWith("/") && !name.endsWith("/")) {
        name += "/";
    }
    return String(_path)+fname;
```

Reply:
@vortigont Your change loses `String name` declaration, and then doesn't use name in return. BUG.

Reply:
right, it's the `name` in second condition. my bad, I should have my glasses on :) tnx!